src/app/clusters/ota-server/ota-server.cpp

```diff
@@ -82,5 +141,31 @@ bool emberAfOtaSoftwareUpdateServerClusterQueryImageCallback(
     /* TYPE WARNING: array array defaults to */ uint8_t * protocolsSupported, uint8_t * location, uint8_t clientCanConsent,
     chip::ByteSpan metadataForServer)
 {
-    return false;
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    ChipLogDetail(Zcl, "OTA Server received QueryImage");
+
+    const uint8_t locationLen = emberAfStringLength(location);
+    if ((locationLen != kLocationParamLength) || (metadataForServer.size() > kMaxMetadataLen))
```

Comment:
Why is this check there: `locationLen != kLocationParamLength` ? Are variable length strings not yet supported in commands?

Reply:
CHAR_STRING is not yet supported in xml. See #6112

Reply:
Another option would be to use OCTET_STRING for now, which is supported (this is suggested in #6805)

Reply:
Please add a TODO and open an issue to track making this right once CHAR_STRING is implemented.

Reply:
Filed #7112